taf/tests/test_updater/test_update_library/test_clone_library.py

```diff
@@ -7,6 +8,23 @@
 )
 
 
+@pytest.mark.parametrize(
+    "library_with_dependencies",
+    [
+        {
+            "targets_config": [{"name": "target1"}, {"name": "target2"}],
+            "dependencies_config": [
```

Comment:
I think that this is a good start. In `conftest`, instead of iterating over an array of two namespaces, we want to iterate over these dependencies. We want to also be able to specify a `targets_config` inside every `dependencies_config` entry. So, it should also be possible to define which targets each referenced auth repository (dependency) has